bundles/org.eclipse.equinox.p2.transport.ecf/src/org/eclipse/equinox/internal/p2/transport/ecf/RepositoryTransport.java

```diff
@@ -268,6 +302,33 @@ public long getLastModified(URI toDownload, IProgressMonitor monitor)
 		throw new AuthenticationFailedException();
 	}
 
+	private URI getSecureLocation(URI location) throws ProvisionException {
```

Comment:
I see one problem with doing the check here is that it will only apply to this specific implementation of a transport, so if there are alternative repository transports used the check is not performed anymore!
Would it be possible to move this from the transport to the application layer so "unsafe" URIs not even reach the transport?